project/FrgaalJavaCompiler.scala

```diff
@@ -78,6 +86,15 @@ object FrgaalJavaCompiler {
     target: String
   ): Boolean = {
     val (jArgs, nonJArgs) = options.partition(_.startsWith("-J"))
+    var noLimitModules = false
+    val strippedJArgs: ArrayBuffer[String] = ArrayBuffer()
+    for (strippedJArg <- jArgs.map(_.stripPrefix("-J"))) {
+      if (strippedJArg == "--no-limit-modules") {
```

Comment:
I am afraid not use `--limit-modules`. It may only work because the `java.compiler`, `jdk.compiler` & co. modules from JDK17 are _close enough_ to the modules needed by Frgaal.

Rather than that I'd like to understand what JDK modules are missing to run annotation processors properly?

Reply:
I have tried removing `--no-limit-modules` in the most recent version of the PR - 7df60c5346f11f7baaeb55727243bd1290519065, but it does not work. Even if I add `org.graalvm.locator` to the list of `--limit-modules`, the `BenchProcessor` fails on `NoClassDefFoundError` in `org.enso.pkg.COnfig#fromYaml` with stack trace:
```
"main@1" prio=5 tid=0x1 nid=NA runnable
  java.lang.Thread.State: RUNNABLE
	  at cats.syntax.EitherObjectOps$.catchNonFatal$extension(either.scala:390)
	  at io.circe.yaml.Parser.parseSingle(Parser.scala:39)
	  at io.circe.yaml.Parser.parse(Parser.scala:29)
	  at io.circe.yaml.parser.package$.parse(package.scala:18)
	  at org.enso.pkg.Config$.fromYaml(Config.scala:256)
	  at org.enso.pkg.PackageManager.$anonfun$loadPackage$2(Package.scala:317)
	  at org.enso.pkg.PackageManager$$Lambda$288/0x00000007c05073f8.apply(Unknown Source:-1)
	  at scala.util.Using$.$anonfun$apply$1(Using.scala:262)
	  at scala.util.Using$.apply(Using.scala:113)
	  at org.enso.pkg.PackageManager.readConfig$1(Package.scala:317)
	  at org.enso.pkg.PackageManager.loadPackage(Package.scala:322)
	  at org.enso.interpreter.runtime.EnsoContext.lambda$initialize$2(EnsoContext.java:146)
	  at org.enso.interpreter.runtime.EnsoContext$$Lambda$286/0x00000007c0506648.apply(Unknown Source:-1)
	  at java.util.Optional.map(Optional.java:260)
	  at org.enso.interpreter.runtime.EnsoContext.initialize(EnsoContext.java:143)
	  at org.enso.interpreter.EnsoLanguage.initializeContext(EnsoLanguage.java:156)
	  at org.enso.interpreter.EnsoLanguage.initializeContext(EnsoLanguage.java:62)
	  at com.oracle.truffle.api.TruffleLanguage$Env.postInit(TruffleLanguage.java:3588)
	  at com.oracle.truffle.api.LanguageAccessor$LanguageImpl.postInitEnv(LanguageAccessor.java:287)
	  at com.oracle.truffle.polyglot.PolyglotLanguageContext.ensureInitialized(PolyglotLanguageContext.java:719)
	  at com.oracle.truffle.polyglot.PolyglotContextImpl.getBindings(PolyglotContextImpl.java:1217)
	  at com.oracle.truffle.polyglot.PolyglotContextDispatch.getBindings(PolyglotContextDispatch.java:98)
	  at org.graalvm.polyglot.Context.getBindings(Context.java:564)
	  at org.enso.benchmarks.processor.BenchProcessor.getModule(BenchProcessor.java:137)
	  at org.enso.benchmarks.processor.BenchProcessor.process(BenchProcessor.java:117)
	  at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1024)
	  at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:940)
	  at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$2300(JavacProcessingEnvironment.java:112)
	  at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1268)
	  at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1383)
	  at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1246)
	  at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:925)
	  at com.sun.tools.javac.main.Main.compile(Main.java:317)
	  at com.sun.tools.javac.main.Main.compile(Main.java:176)
	  at com.sun.tools.javac.Main.compile2(Main.java:76)
	  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-1)
	  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	  at java.lang.reflect.Method.invoke(Method.java:568)
	  at org.frgaal.Main.lambda$compile$0(Main.java:24)
	  at org.frgaal.Main$$Lambda$24/0x00000007c00c0900.run(Unknown Source:-1)
	  at org.frgaal.AbstractMain.doRun(AbstractMain.java:27)
	  at org.frgaal.Main.compile(Main.java:21)
	  at org.frgaal.Main.main(Main.java:17)
```

For now, I am leaving the `--no-limit-modules` there, with the exception that there will be a documented constant in `FrgaalJavaCompiler.scala` instead of hardcoded "-J--no-limit-modules".

Reply:
Refactored to a documented constant in `FrgaalJavaCompiler.scala` in 6676b8cde22e31eb569c5bd0bdffa63eff6165cf